include/triSYCL/detail/fiber/pooled_work_stealing.hpp

```diff
@@ -107,11 +107,8 @@ class pooled_work_stealing : public boost::fibers::algo::algorithm {
     }
 
 
-  ~pooled_work_stealing() {
-    // Wait for all thread of the pool such that pointers in pool_ctx_
-    // stay valid while still in use
-    pool_ctx_->barrier_.wait();
-  }
+  // This will not be called until all other fibers are ready.
+  ~pooled_work_stealing() = default;
```

Comment:
Do we need this line at all?
I guess we will have a default destructor anyway...

Reply:
this isn't needed since it is defaulted but i kept it to add a comment on it.
can be removed if you wish.

Reply:
That sounds good.